src/rust/catnap/mod.rs

```diff
@@ -519,7 +529,10 @@ fn pack_result(rt: &DemiRuntime, result: OperationResult, qd: QDesc, qt: u64) ->
             qr_value: unsafe { mem::zeroed() },
         },
         OperationResult::Accept((new_qd, addr)) => {
+            #[cfg(target_os = "linux")]
             let saddr: SockAddr = linux::socketaddrv4_to_sockaddr(&addr);
+            #[cfg(target_os = "windows")]
+            let saddr: SockAddr = convert_socket_addr_v4_to_sockaddr(&addr);
```

Comment:
Could you around this logic around a function so that we don't need to have these checks outside the `pal` component?

Reply:
Sure, good idea. Will update.